### PR TITLE
feat: add helper functions to manager finalizer

### PR DESCRIPTION
**Pull request by nanjingfm**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [`spec` PR link](https://github.com/katanomi/spec) included
- [x] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [x] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [x] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

### CODE REVIEW

File: apis/selection/v1alpha1/base_filter.go [hunk 1]
```diff
@@ -46,6 +50,38 @@ type BaseFilter struct {
 	Refs []corev1.ObjectReference `json:"refs,omitempty"`
 }
 
+func (p *BaseFilter) MatchObject(obj client.Object) bool {
```
Comment (l-qing) [inline apis/selection/v1alpha1/base_filter.go:53]: 公开函数加点注释？

File: apis/selection/v1alpha1/base_filter.go [hunk 2]
```diff
@@ -46,6 +50,38 @@ type BaseFilter struct {
...
+			if metav1alpha1.IsTheSameObject(ref, objRef) {
+				isExcept = true
+				break
+			}
+		}
+	}
+	if !isExcept {
+		return false
+	}
+	if p.Filter != nil && !p.Filter.MatchExact(obj) {
```
Comment (l-qing) [inline apis/selection/v1alpha1/base_filter.go:79]: 一个 return 也行？

File: finalizer/finalizer.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			"namespacedName", client.ObjectKeyFromObject(o),
+			"finalizerKey", finalizerKey,
+		)
+	}
+	return err
+}
+
+// RemoveFinalizer removes a finalizer from the object
+func RemoveFinalizer(ctx context.Context, clt client.Client, o client.Object, finalizerKey string,
+	callback func() error) error {
```
Comment (l-qing) [inline finalizer/finalizer.go:66]: callback 参数要不要加个 ctx？

Comment (nanjingfm) [reply finalizer/finalizer.go:66]: ctx就是调用的地方传递过来的，可以直接拿到ctx。

函数内并不会修改ctx

**Review by l-qing — APPROVED** [review]

**Pull request merged**